### PR TITLE
[Hexagon] Add doc on TVM - Hexagon RPC flow

### PR DESCRIPTION
**Pull request by masahi**
[Rendered](https://github.com/masahi/tvm/blob/tvm-hex-rpc-doc/tests/python/contrib/test_hexagon/README_RPC.md)

This PR adds documentation on the complex interaction between TVM and Hexagon RPC. 

@kparzysz-quic @mehrdadh @areusch @tmoreau89

### CODE REVIEW

**Comment by masahi** [comment]
@kparzysz-quic As my doc shows, `LocalSession` is not involved in TVM - Hexagon RPC. So we can delete 
https://github.com/apache/tvm/blob/cb7f773218c26c3092cc275ef28e71839ebd18e3/cmake/modules/Hexagon.cmake#L249

Can I go ahead and do that?

**Comment by kparzysz-quic** [comment]
You can try that, but when I did, it caused failures.  Make sure that once you delete it, it still works.

File: tests/python/contrib/test_hexagon/README_RPC.md [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+## Initialization: Setting up Android and establishing connection between x86 host and android
+
+What’s happening during the launcher initialization at [https://github.com/apache/tvm/blob/2b35cfd6ddb73afecd3f550f33881e1fdc7c3267/tests/python/contrib/test_hexagon/rpc/test_launcher.py#L98-L107](https://github.com/apache/tvm/blob/2b35cfd6ddb73afecd3f550f33881e1fdc7c3267/tests/python/contrib/test_hexagon/rpc/test_launcher.py#L98-L107) ?
+
+```cpp
+launcher = HexagonLauncher(serial_number=android_serial_number)
+launcher.android_run_rpc(rpc_tracker_host=tvm_tracker_host, rpc_tracker_port=tvm_tracker_port)
+launcher.hexagon_setup()
+remote_kw = {...}
+launcher.hexagon_session_setup(remote_kw)
```
Comment (kparzysz-quic) [inline tests/python/contrib/test_hexagon/README_RPC.md:43]: This is out of date.

Comment (masahi) [reply tests/python/contrib/test_hexagon/README_RPC.md]: Thanks updated for the latest code.

**Comment by masahi** [comment]
> You can try that, but when I did it, it caused failures. Make sure that once you delete it, it still works.
> 
> Edit: please check both, hardware and simulator.

Ok, I'll do that later. I wonder why you got the failure. Karl (don't know his github) got a call trace on the Hexagon side which showed: 
```
* thread #2, stop reason = breakpoint 7.1
  * frame #0: 0xfdf4bc30 libhexagon_rpc_skel.so`tvm::runtime::hexagon::HexagonDeviceAPIv2::CopyDataFromTo(DLTensor*, DLTensor*, void*)
    frame #1: 0xfde9b4a4 libhexagon_rpc_skel.so`TVMDeviceCopyDataFromTo + 68
    frame #2: 0xfde78b20 libhexagon_rpc_skel.so`tvm::runtime::MinRPCServer<tvm::runtime::hexagon::HexagonIOHandler>::HandleCopyToRemote() + 272
    frame #3: 0xfde785e4 libhexagon_rpc_skel.so`tvm::runtime::MinRPCServer<tvm::runtime::hexagon::HexagonIOHandler>::ProcessOnePacket() + 212
    frame #4: 0xfde777b8 libhexagon_rpc_skel.so`hexagon_rpc_send + 52
    frame #5: 0x00180970 fastrpc_shell_unsigned_3`mod_table_invoke + 528
    frame #6: 0x00171b6c fastrpc_shell_unsigned_3`fastrpc_invoke_dispatch + 556
```
So there is no `LocalSession` involved on the Hexagon side.

**Comment by masahi** [comment]
@kparzysz-quic please review

**Review by kparzysz-quic — APPROVED** [review]

**Pull request merged**